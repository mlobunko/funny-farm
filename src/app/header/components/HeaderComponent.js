import React from 'react';
import PropTypes from 'prop-types';

import {
  StyledHeaderComponent,
  StyledH1,
  StyledLeft,
  StyledMusicIcon,
  StyledRight
} from './styles';

export const HeaderComponent = ({ isMusicOn, toggleMusic }) => (
  <StyledHeaderComponent>
    <StyledLeft />
    <StyledH1>Funny Farm</StyledH1>
    <StyledRight>
      <StyledMusicIcon onClick={toggleMusic} isMusicOn={isMusicOn} />
    </StyledRight>
  </StyledHeaderComponent>
);

export default HeaderComponent;

HeaderComponent.prototypes = {
  isMusicOn: PropTypes.bool.isRequired,
  toggleMusic: PropTypes.func.isRequired
};
