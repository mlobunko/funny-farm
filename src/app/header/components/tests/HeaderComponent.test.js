import React from 'react';
import { shallow } from 'enzyme';

import { HeaderComponent } from '../HeaderComponent';

const toggleMusic = jest.fn();
const isMusicOn = true;

describe('render', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <HeaderComponent isMusicOn={isMusicOn} toggleMusic={toggleMusic} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('StyledMusicIcon', () => {
  it('find toggleMusic in onClick', () => {
    const wrapper = shallow(
      <HeaderComponent isMusicOn={isMusicOn} toggleMusic={toggleMusic} />
    );
    expect(wrapper.find('StyledMusicIcon').prop('onClick')).toEqual(
      toggleMusic
    );
  });
});
