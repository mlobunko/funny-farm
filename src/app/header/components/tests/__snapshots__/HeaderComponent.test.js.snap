// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledHeaderComponent>
  <StyledLeft />
  <StyledH1>
    Funny Farm
  </StyledH1>
  <StyledRight>
    <StyledMusicIcon
      isMusicOn={true}
      onClick={[Function]}
    />
  </StyledRight>
</StyledHeaderComponent>
`;
